src/main/java/bdv/fx/viewer/render/RenderUnit.java

```diff
@@ -339,7 +346,16 @@ public void paint() {
 					null
 			);
 
-			renderTarget.drawOverlays(image::set);
+			if (renderedScreenScaleIndex != -1)
+			{
+				final Double renderedScreenScale;
+				synchronized (RenderUnit.this)
```

Comment:
Why synchronize on `RenderUnit.this` here? Is it becasue `screenScales` may change during the call to `screenScales[screenScaleindex]`?